src/java.desktop/share/classes/javax/swing/SwingWorker.java

```diff
@@ -566,7 +565,7 @@ public final boolean isCancelled() {
      * {@inheritDoc}
      */
     public final boolean isDone() {
-        return future.isDone();
+        return future.isDone() && state == StateValue.DONE;
```

Comment:
```suggestion
        return state == StateValue.DONE;
```
I guess we can remove the call to `future.isDone()` altogether. Either both conditions are true, or both are false.

Reply:
It will cause JCK issue..